### PR TITLE
Implement entry and persistence of custom base node in console wallet

### PR DESCRIPTION
**Pull request by philipr-za**
## Description
This PR adds the functionality to allow a user to specify a custom base node public key and address in the Network tab of the console wallet. If the public key and address can be parsed into a Comms Public Key and valid Net address this bae node is set to be used by all the wallet services and is saved into the wallet’s client key-value store.

On startup if there is a valid public key and address stored in the wallets client key-value store this is used as the base node for the wallet to connect to. If these values don’t exist or cannot be parsed into a valid public key or net address the wallet will fallback to the base node specified in the config.

A function is also added to clear the custom base node which, after a confirmation dialog, will delete the custom data from the database and revert to the config specified base node.

## How Has This Been Tested?
Manually tested

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_console_wallet/src/ui/app.rs [hunk 1]
```diff
@@ -55,29 +63,55 @@ pub struct App<B: Backend> {
...
-        network: Network,
-        base_node: Peer,
-    ) -> Self
-    {
-        // TODO: It's probably better to read the node_identity from the wallet, but that requires
-        // taking a read lock and making this method async, which adds some read/write cycles,
-        // so it's easier to just ask for it right now
-        let app_state = AppState::new(&node_identity, network, wallet, base_node);
+    pub async fn new(title: String, mut wallet: WalletSqlite, network: Network, base_node_config: Peer) -> Self {
+        // Attempt to read a stored custom base node puclic key and address from the wallet database. If this fails we
```
Comment (delta1) [inline applications/tari_console_wallet/src/ui/app.rs]: s/puclic/public/

**Review by delta1 — DISMISSED** [review]
LGTM tested 👍

File: applications/tari_console_wallet/src/ui/app.rs (hunk 1 above)
Comment (philipr-za) [reply applications/tari_console_wallet/src/ui/app.rs]: damnit, my speling is usully purfect

**Review by delta1 — APPROVED** [review]

**Comment by hansieodendaal** [comment]
Certainly needed

**Pull request merged**